hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastService.java

```diff
@@ -59,8 +60,10 @@
     private final DatagramPacket datagramPacketSend;
     private final DatagramPacket datagramPacketReceive;
 
+    private volatile long lastLoggedJoinSerializationFailure;
```

Comment:
Does it have to be `volatile`? As far as I can tell, this field is read & updated from a single thread ([`multicastServiceThread`](https://github.com/vbekiaris/hazelcast/blob/0f62712f53ae6e207e7909f785f5f3c78e960393/hazelcast/src/main/java/com/hazelcast/instance/Node.java#L379-L379)) and its default value `0` will be visible to the subsequently started thread.